src/js/apps/patients/worklist/worklist_app.js

```diff
@@ -151,6 +154,29 @@ export default App.extend({
 
     this.showList();
   },
+  subscribe() {
```

Comment:
Alright I think this whole function needs refactoring..

we should probably just split based on `filterKey` and hardcode types for this one..

also somewhere we can call break this up into another function
```
subscribe() {
  this.listenTo(... () => {
    fetch().then(() => {. // <- one or both of these could be a separate handler..
```

But I think we still want to check the store for the resource.. but if we need to fetch it, as mentioned below.. switch to either the action or flow entity request rather than adding the fetch specifications here for include/fields.

Reply:
I took a crack at refactoring this in https://github.com/RoundingWell/care-ops-frontend/pull/1380/commits/836c7c01aa9d8cd6d86bfbacb59c0977853f5302, let me know what you think.